pkg/controller/masteruserrecord/sync.go

```diff
@@ -0,0 +1,82 @@
+package masteruserrecord
+
+import (
+	"context"
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Synchronizer struct {
+	hostClient        client.Client
+	memberClient      client.Client
+	memberUserAcc     *toolchainv1alpha1.UserAccount
+	recordSpecUserAcc toolchainv1alpha1.UserAccountEmbedded
+	record            *toolchainv1alpha1.MasterUserRecord
+}
+
+func (s *Synchronizer) synchronizeSpec() error {
+	if !reflect.DeepEqual(s.memberUserAcc.Spec, s.recordSpecUserAcc.Spec) {
+		// when UserAccount spec in record is updated - is not same as in member
+		s.memberUserAcc.Spec = s.recordSpecUserAcc.Spec
+		if err := updateStatusConditions(s.hostClient, s.record, toBeNotReady(updatingReason, "")); err != nil {
+			return err
+		}
+		err := s.memberClient.Update(context.TODO(), s.memberUserAcc)
+		if err != nil {
+			return err
```

Comment:
Good point, but I think that the `sync.go` file is not the best place to do it. I made it one function "higher": 788fa1a
As for logging the error - my answer is same - it's logged in the `Reconcile` method and I think that it is more than enough.
BTW, I don't think that it is needed to add the name of the UserAcc to the msg because:
* the UserAcc name is same as MUR, so just target cluster is enough for the condition reason
* the request logger already contains (as a parameter) the name of the MUR, so putting it into the msg is redundant